client/src/components/newSource/index.tsx

```diff
@@ -154,15 +154,23 @@ const NewSourceView = (): React.ReactElement => {
           } catch (e) {
             reject(new Error(`could not parse ${mappingFile.name} as JSON`));
           }
-          await client.mutate({
-            mutation: mCreateSource,
-            variables: {
-              templateName,
-              hasOwner,
-              mapping,
-              name: sourceName
-            }
-          });
+          await client
```

Comment:
why not using try {} catch (err) instead of the old-style .catch ?

Reply:
I'm not sure exactly how the FileReader and this function work together, I broke that once... So I went for the minimal change 😅
But I can try again quickly